File: sample/src/main/java/com/github/adnansm/timelytextview/sample/MainActivity.java
package com.github.adnansm.timelytextview.sample;

import android.app.Activity;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.SeekBar;
import android.widget.Spinner;

import com.github.adnansm.timelytextview.TimelyView;
import com.nineoldandroids.animation.ObjectAnimator;

public class MainActivity extends Activity {
    public static final int DURATION = 1000;
    public static final int NO_VALUE = -1;
    private TimelyView timelyView = null;
    private volatile ObjectAnimator objectAnimator = null;

    private volatile int from = NO_VALUE;
    private volatile int to = NO_VALUE;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        timelyView = (TimelyView) findViewById(R.id.textView1);
        SeekBar seekBar = (SeekBar) findViewById(R.id.seekBar);
        Spinner fromSpinner = (Spinner) findViewById(R.id.fromSpinner);
        Spinner toSpinner = (Spinner) findViewById(R.id.toSpinner);

        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.from_numbers_array, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        fromSpinner.setAdapter(adapter);
        toSpinner.setAdapter(adapter);
        fromSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                from = position - 1;
                if (from != NO_VALUE && to != NO_VALUE) {
                    objectAnimator = timelyView.animate(from, to);
                    objectAnimator.setDuration(DURATION);
                } else {
                    objectAnimator = null;
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
            }
        });
        toSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                to = position - 1;
                if (from != NO_VALUE && to != NO_VALUE) {
                    objectAnimator = timelyView.animate(from, to);
                    objectAnimator.setDuration(DURATION);
                } else {
                    objectAnimator = null;
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });

        seekBar.setMax(DURATION);
        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                if (objectAnimator != null) objectAnimator.setCurrentPlayTime(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

}

File: sample/src/main/java/com/github/adnansm/timelytextview/sample/MainActivity.kt
package com.github.adnansm.timelytextview.sample


import android.app.Activity
import android.os.Bundle
import android.view.Menu
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.ArrayAdapter
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.Spinner
import com.github.adnansm.timelytextview.TimelyView
import android.animation.ObjectAnimator

class MainActivity : Activity() {
    private var timelyView: TimelyView? = null

    @Volatile
    private var objectAnimator: ObjectAnimator? = null

    @Volatile
    private var from = NO_VALUE

    @Volatile
    private var to = NO_VALUE
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        timelyView = findViewById<View>(R.id.textView1) as TimelyView
        val seekBar = findViewById<View>(R.id.seekBar) as SeekBar
        val fromSpinner = findViewById<View>(R.id.fromSpinner) as Spinner
        val toSpinner = findViewById<View>(R.id.toSpinner) as Spinner
        val adapter = ArrayAdapter.createFromResource(
            this,
            R.array.from_numbers_array,
            android.R.layout.simple_spinner_item
        )
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        fromSpinner.adapter = adapter
        toSpinner.adapter = adapter
        fromSpinner.onItemSelectedListener = object : OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                from = position - 1
                if (from != NO_VALUE && to != NO_VALUE) {
                    objectAnimator = timelyView!!.animate(from, to)
                    objectAnimator!!.duration = DURATION.toLong()
                } else {
                    objectAnimator = null
                }
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {}
        }
        toSpinner.onItemSelectedListener = object : OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                to = position - 1
                if (from != NO_VALUE && to != NO_VALUE) {
                    objectAnimator = timelyView!!.animate(from, to)
                    objectAnimator!!.duration = DURATION.toLong()
                } else {
                    objectAnimator = null
                }
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {}
        }
        seekBar.max = DURATION
        seekBar.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                if (objectAnimator != null) objectAnimator!!.currentPlayTime = progress.toLong()
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    companion object {
        const val DURATION = 1000
        const val NO_VALUE = -1
    }
}